contracts/core/modules/IssuanceOrderModule.sol

```diff
@@ -236,109 +235,32 @@ contract IssuanceOrderModule is
-
-            // Verify exchange address is registered
-            require(
-                exchangeWrapper != address(0),
-                "IssuanceOrderModule.executeExchangeOrders: Invalid or disabled Exchange address"
-            );
-
-            // Read the order body based on order data length info in header plus the length of the header (128)
-            uint256 exchangeDataLength = header.orderDataBytesLength.add(128);
-            bytes memory bodyData = LibBytes.slice(
-                _orderData,
-                scannedBytes.add(128),
-                scannedBytes.add(exchangeDataLength)
-            );
-
-            // Transfer maker token to Exchange Wrapper to execute exchange orders
-            // Using maker token from signed issuance order to prevent malicious encoding of another maker token
-            ITransferProxy(transferProxy).transfer(
-                _makerTokenAddress,
-                header.makerTokenAmount,
-                _makerAddress,
-                exchangeWrapper
-            );
-
-            // Call Exchange
-            address[] memory componentFillTokens = new address[](header.orderCount);
-            uint256[] memory componentFillAmounts = new uint256[](header.orderCount);
-            (componentFillTokens, componentFillAmounts) = IExchangeWrapper(exchangeWrapper).exchange(
-                _makerAddress,
-                msg.sender,
-                _makerTokenAddress,
-                header.makerTokenAmount,
-                header.orderCount,
-                bodyData
-            );
-
-            // Transfer component tokens from wrapper to vault
-            coreInstance.batchDepositModule(
-                exchangeWrapper,
-                _makerAddress,
-                componentFillTokens,
-                componentFillAmounts
-            );
-
-            // Update scanned bytes with header and body lengths
-            scannedBytes = scannedBytes.add(exchangeDataLength);
-            makerTokenUsed = makerTokenUsed.add(header.makerTokenAmount);
-        }
-
-        return makerTokenUsed;
-    }
-
-    /**
-     * Validate order params are still valid
+     * Validate order params are still valid and return amount that can still be
+     * executed (after previous fills and cancels accounted for)
      *
      * @param  _order              IssuanceOrder object containing order params
      * @param  _executeQuantity    Quantity of Set to be filled
+     * @return  uint256            The executable amount of the cancel or fill
      */
     function validateOrder(
         OrderLibrary.IssuanceOrder _order,
         uint256 _executeQuantity
     )
-        private
+        public
         view
+        returns (uint256)
     {
         // Declare set interface variable
         ISetToken set = ISetToken(_order.setAddress);
 
+        // Determine amount to execute
+        uint256 closedOrderAmount = orderFills[_order.orderHash].add(orderCancels[_order.orderHash]);
```

Comment:
What is the purpose of `validateOrder`? As I read the function name, I understand it to "validate an order" - which is making assertions that either validate or invalidate the order. I don't expect it to do anything else.

Checking how much of a Set is executable is separate from validating an order, so it would make sense to put into another function. I also wouldn't expect (as a user/developer) to receive the executable amount from calling validateOrder. I would expect a boolean, a revert/no revert.

Adding this additional functionality makes this function do more than validate an order (doing more than a single thing) and it is logically misplaced.

Reply:
Well we can change the function name if you think it's misleading, I do think that part of validating an order is knowing how much you're gonna get back. But I still don't see why this needs to be split into separate functions. It's adding unnecessary gas, for little clarity.

Reply:
As I think about it more, `validateOrder` should only take the `OrderLibrary.IssuanceOrder` parameter. Whether the order is fillable with the `executeQuantity` and how much it can be filled is a different operation. Thus, the function would truly be validating whether an order is valid.

Then we can write a different function called `validateOrderExecutable` that takes in the `orderHash` and `executeQuantity` which can check whether the order is executable as well as return the executable amount.

Reply:
This should not be in `validateOrder`. Is there a stack size issue here?